MekHQ/src/mekhq/gui/dialog/AddOrEditKillEntryDialog.java

```diff
@@ -47,29 +48,44 @@ public class AddOrEditKillEntryDialog extends JDialog {
     private int operationType;
     private Kill kill;
     private LocalDate date;
+    private int missionId;
+    private int scenarioId;
+    private Campaign campaign;
 
     private JButton btnClose;
     private JButton btnOK;
     private JLabel lblKill;
     private JTextField txtKill;
     private JLabel lblKiller;
     private JTextField txtKiller;
+    private JLabel lblDate;
     private JButton btnDate;
+    private JLabel lblMissionId;
+    private MMComboBox<Mission> cboMissionId;
+    private ArrayList<Integer> missionIdList;
+    private JLabel lblScenarioId;
+    private MMComboBox<String> cboScenarioId;
+    private ArrayList<Integer> scenarioIdList;
 
-    public AddOrEditKillEntryDialog(JFrame parent, boolean modal, UUID killerPerson, String killerUnit, LocalDate entryDate) {
-        this(parent, modal, ADD_OPERATION, new Kill(killerPerson, "?", killerUnit, entryDate));
+    public AddOrEditKillEntryDialog(JFrame parent, boolean modal, UUID killerPerson, String killerUnit, LocalDate entryDate, Campaign campaign) {
+        // We default missionId & scenarioId to 0 when adding new kills
+        this(parent, modal, ADD_OPERATION, new Kill(killerPerson, "?", killerUnit, entryDate, 0, 0), campaign);
```

Comment:
If missionId and ScenarioId were initialized to 0, a simpler constructor could be used here.

Reply:
So, I looked into this and (assuming I'm not missing your point entirely, a non-zero chance) I don't think the functionality is improved by using a constructor here. This is the only situation where we're using the default value of `0` and that's to ensure that when the manual 'add kill' dialog triggers it's not at risk of encountering a `null` value for Scenario of Mission ID, before the new value is set. The moment the manual Kill is confirmed that default value is only used if no other value is set.

`Kill.java` already has `missionID` and `scenarioID` initializing at 0.